### ISSUE TITLE
TypeError while trying to clear output

### ISSUE THREAD

**Issue by alexandrekl** [spyder-ide/spyder#15112]
### What steps will reproduce the problem?

opened a notebook > Kernet > restart kernel and clear all ouput

### Traceback
```python-traceback
  File "/Users/aligo/opt/anaconda3/lib/python3.7/site-packages/qtconsole/base_frontend_mixin.py", line 138, in _dispatch
    handler(msg)
  File "/Users/aligo/opt/anaconda3/lib/python3.7/site-packages/qtconsole/frontend_widget.py", line 571, in _handle_shutdown_reply
    self._handle_kernel_restarted(died=False)
TypeError: _handle_kernel_restarted() got an unexpected keyword argument 'died'
```

## Versions

* Spyder version: 4.2.3 
* Python version: 3.7.6
* Qt version: 5.9.6
* PyQt5 version: 5.9.2
* Operating System: Darwin 20.3.0

### Dependencies

```

# Mandatory:
applaunchservices >=0.1.7       :  0.2.1 (OK)
atomicwrites >=1.2.0            :  1.4.0 (OK)
chardet >=2.0.0                 :  4.0.0 (OK)
cloudpickle >=0.5.0             :  1.6.0 (OK)
diff_match_patch >=20181111     :  20200713 (OK)
intervaltree >=3.0.2            :  3.1.0 (OK)
IPython >=7.6.0                 :  7.22.0 (OK)
jedi =0.17.2                    :  0.17.2 (OK)
jsonschema >=3.2.0              :  3.2.0 (OK)
keyring >=17.0.0                :  22.3.0 (OK)
nbconvert >=4.0                 :  6.0.7 (OK)
numpydoc >=0.6.0                :  1.1.0 (OK)
parso =0.7.0                    :  0.7.0 (OK)
pexpect >=4.4.0                 :  4.8.0 (OK)
pickleshare >=0.4               :  0.7.5 (OK)
psutil >=5.3                    :  5.8.0 (OK)
pygments >=2.0                  :  2.8.1 (OK)
pylint >=1.0                    :  2.7.4 (OK)
pyls >=0.36.2;<1.0.0            :  0.36.2 (OK)
pyls_black >=0.4.6              :  0.4.6 (OK)
pyls_spyder >=0.3.2             :  0.3.2 (OK)
qdarkstyle >=2.8                :  3.0.2 (OK)
qtawesome >=0.5.7               :  1.0.2 (OK)
qtconsole >=5.0.1               :  5.0.3 (OK)
qtpy >=1.5.0                    :  1.9.0 (OK)
rtree >=0.8.3                   :  0.9.4 (OK)
setuptools >=39.0.0             :  52.0.0.post20210125 (OK)
sphinx >=0.6.6                  :  3.5.3 (OK)
spyder_kernels >=1.10.2;<1.11.0 :  1.10.2 (OK)
textdistance >=4.2.0            :  4.2.1 (OK)
three_merge >=0.1.1             :  0.1.1 (OK)
watchdog >=0.10.3;<2.0.0        :  1.0.2 (OK)
zmq >=17                        :  20.0.0 (OK)

# Optional:
cython >=0.21                   :  0.29.22 (OK)
matplotlib >=2.0.0              :  3.3.4 (OK)
numpy >=1.7                     :  1.19.2 (OK)
pandas >=1.1.1                  :  1.2.3 (OK)
scipy >=0.17.0                  :  1.6.2 (OK)
sympy >=0.7.3                   :  1.7.1 (OK)

# Spyder plugins:
spyder_notebook                 :  0.3.2 (OK)
```

**Comment by steff456** [comment]
Hi @alexandrekl,

Thanks for reporting, @impact27 do you know what is happening here?

**Comment by impact27** [comment]
There is a missing argument. I fixed it here: https://github.com/spyder-ide/spyder/pull/15135

**Comment by steff456** [comment]
Thanks a lot @impact27!

**Comment by ccordoba12** [comment]
Thanks @impact27 for your help!

@alexandrekl, the fix will be available in our versions 4.2.6 (to released in a month or so) and 5.0.1 (to be released in the next couple of days).

If you don't want to update to Spyder 5 (which is fine), please be patient until 4.2.6 is out.
